library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -71,21 +71,50 @@ internal class TransactionPayloadFragment :
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+
+        val result = mutableListOf<TransactionPayloadItem>()
+        val adapter = TransactionBodyAdapter(result)
+        payloadBinding.responseRecyclerView.adapter = adapter
+        payloadBinding.responseRecyclerView.setHasFixedSize(true)
+
+        viewModel.encodeUrl.observe(
+            viewLifecycleOwner,
+            Observer { encode ->
+                if (adapter.itemCount == 0) {
```

Comment:
Why do you need this check if you also do the check for transaction value below?

Reply:
Definitely something I should have commented. There are cases in which `transaction` is not null here (e.g., configuration change), so you end up notifying the adapter twice: once here and once when observing `transaction`. Not a big deal, but it's unnecessary work.

I tried to think of a way not to have this here and the only thing I could think of is having a separate `transaction` LiveData that under the hood is just a MediatorLiveData that combines `transaction` and `encodeUrl`, taking care of notifying observers only once the first time it's observed. It didn't look pretty and it was more convoluted than this simple check, which is just a micro-optimization and not something required. With or without this check everything would work just fine. But I don't really like the fact that this trick only works if you start observing this LiveData before the other.